packages/belos/src/BelosSolverManager.hpp

```diff
@@ -239,8 +328,27 @@ namespace Details {
     public SolverManager<ScalarType, MV, OP> {
   public:
     RealSolverManager () {
-      TEUCHOS_TEST_FOR_EXCEPTION( true, std::logic_error,
-        "This solver is not implemented for complex ScalarType." );
+      // TODO: How should we handle Tpetra complex for DII registration?
+      // I allow the construction so automatic registration of Tpetra complex can
+      // proceed in a generic way and build these 'dummy' classes.
+      // But then this class will not actually get used for anything.
+      // I'll need to discuss whether we want to have a universal
```

Comment:
Is there any reason why you can't just create the SolverManager and skip the whole real/complex distinction?  The point of `RealSolverManager` etc. was so that `Belos::SolverFactory` would "just work" without needing specializations for real vs. complex.  I think you could just rely on that.

Reply:
Same as above comment - now I just register all the managers for complex without any special casing. Belos and Tpetra are handled in a similar way. But this formatting has all changed quite a bit now that we do everything in a single cpp for each lib.